integration_test.go

```diff
@@ -161,3 +161,72 @@ func TestAllowedRegexpProxy(t *testing.T) {
 	// TODO(oskar) - gracefully stop proxy by the end of the test
 	es.Stop(ctx)
 }
+
+func TestBasicAuthRoundtrip(t *testing.T) {
+	tc := newTestClient()
+
+	echoServerAddr := "localhost:3010"
+	es := newEchoServer(echoServerAddr)
+
+	ctx := context.Background()
```

Comment:
Seems this `ctx` isn't used anywhere?

Reply:
It is used at the end with:
```
	es.Stop(ctx)
```

However now we also stop the proxyServers explicitly with:
```
	es.Stop(ctx)
	authProxyServer.Shutdown(ctx)
	signingProxyServer.Shutdown(ctx)
```